### PR TITLE
executor: fix IsPointGet judgment condition (#10278)

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

fix IsPointGet judgment condition

### What is changed and how it works?

Cherry pick from https://github.com/pingcap/tidb/pull/10278
(only the fix part, because the master branch evolves and differ from the release-2.0 branch)

### CODE REVIEW

**Comment by jackysp** [comment]
Please fix CI.

File: executor/adapter.go [hunk 1]
```diff
@@ -424,9 +424,6 @@ func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx sessionctx.Context, p plan.Plan
 	case *plan.PhysicalIndexReader:
 		indexScan := v.IndexPlans[0].(*plan.PhysicalIndexScan)
 		return indexScan.IsPointGetByUniqueKey(ctx.GetSessionVars().StmtCtx), nil
-	case *plan.PhysicalIndexLookUpReader:
```
Comment (zz-jason) [inline executor/adapter.go]: index lookup is not allowed to use point get?

Comment (jackysp) [reply executor/adapter.go]: yes

Comment (zz-jason) [reply executor/adapter.go]: It's more like a workaround other than a bugfix ..

**Comment by jackysp** [comment]
merged, but no test...

File: executor/adapter.go (hunk 1 above)
Comment (coocood) [reply executor/adapter.go]: This point get is not point get executor, it's using MaxTS.

**Comment by tiancaiamao** [comment]
Yes, no test @jackysp 
> (only the fix part, because the master branch evolves and differ from the release-2.0 branch)

File: executor/adapter.go (hunk 1 above)
Comment (tiancaiamao) [reply executor/adapter.go]: It's a bugfix... maybe the function name is a bit confusing...
index lookup is not allowed to use max ts and this function checks whether to use max ts
https://github.com/tiancaiamao/tidb/blob/ab1abb1938782f4bd69d15650489f9074e26ade6/executor/adapter.go#L312
@zz-jason

Comment (zz-jason) [reply executor/adapter.go]: got it, thanks

**Pull request merged**